DataStructures/HashTable.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using LinkedList;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hash_Table
+{
+    public class HashTable : IHashTable
+    {
+        private object[] _array;
+        public int Length => _array.Length;
+
+
+        public HashTable()
+        {
+            _array = new object[3];
+        }
+
+        public int Hash(object key)
+        {
+            return key.GetHashCode() & 0x7fffffff % 10;
+        }
+        public bool Contains(object key)
+        {
+            if (_array[Hash(key)] == null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public void Add(object key, object value)
+        {
+            int index = Hash(key);
+            if (index > _array.Length - 1)
+            {
+                object[] temp = new object[index + 1];
+                for (int i = 0; i <= _array.Length - 1; i++)
+                {
+                    temp[i] = _array[i];
+                }
+
+                _array = temp;
+            }
+
+            if (_array[index] != null)
+            {
+                throw new DuplicateNameException();
+            }
+
+            _array[index] = value;
+        }
+
+        public object this[object key]
```

Comment:
There are bugs as well as the code can be improved. Try to resolve it on your own. Let me know if you need any help.